notion-server/src/eth_magic.rs

```diff
@@ -0,0 +1,165 @@
+use std::default;
+use std::collections::{HashMap};
+
+#[derive(Debug)]
+pub struct EthMagicManager {
```

Comment:
While I enjoy this name, I don't think it is the most descriptive for what this struct does. The words `Magic` and `Manager` don't add useful information. I'd go with just `Ethereum` or `EthereumSimulation` if you prefer longer names.

Reply:
I like `EthereumSimulation` -- I'll change to that.